pkg/kubectl/cmd/edit.go

```diff
@@ -0,0 +1,415 @@
+	}
+
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	defaultVersion := cmdutil.OutputVersion(cmd, clientConfig.Version)
+	results := editResults{}
+	for {
+		obj, err := resource.AsVersionedObject(infos, false, defaultVersion)
+		if err != nil {
+			return preservedFile(err, results.file, cmd.Out())
+		}
+
+		// TODO: add an annotating YAML printer that can print inline comments on each field,
+		//   including descriptions or validation errors
+
+		// generate the file to edit
+		buf := &bytes.Buffer{}
+		if err := results.header.WriteTo(buf); err != nil {
+			return preservedFile(err, results.file, cmd.Out())
+		}
+		if err := printer.PrintObj(obj, buf); err != nil {
+			return preservedFile(err, results.file, cmd.Out())
+		}
+		original := buf.Bytes()
+
+		// launch the editor
+		edit := editor.NewDefaultEditor()
+		edited, file, err := edit.LaunchTempFile("kubectl-edit-", ext, buf)
+		if err != nil {
+			return preservedFile(err, results.file, cmd.Out())
+		}
+
+		// cleanup any file from the previous pass
+		if len(results.file) > 0 {
+			os.Remove(results.file)
+		}
+
+		glog.V(4).Infof("User edited:\n%s", string(edited))
+		lines, err := hasLines(bytes.NewBuffer(edited))
+		if err != nil {
+			return preservedFile(err, file, cmd.Out())
+		}
+		if bytes.Equal(original, edited) {
+			if len(results.edit) > 0 {
+				preservedFile(nil, file, cmd.Out())
+			} else {
+				os.Remove(file)
+			}
+			fmt.Fprintln(cmd.Out(), "Edit cancelled, no changes made.")
+			return nil
+		}
+		if !lines {
+			if len(results.edit) > 0 {
+				preservedFile(nil, file, cmd.Out())
+			} else {
+				os.Remove(file)
+			}
+			fmt.Fprintln(cmd.Out(), "Edit cancelled, saved file was empty.")
+			return nil
+		}
+
+		results = editResults{
+			file: file,
+		}
+
+		// parse the edited file
+		updates, err := rmap.InfoForData(edited, "edited-file")
+		if err != nil {
+			results.header.reasons = append(results.header.reasons, editReason{
+				head: fmt.Sprintf("The edited file had a syntax error: %v", err),
+			})
+			continue
+		}
+
+		visitor := resource.NewFlattenListVisitor(updates, rmap)
+
+		// need to make sure the original namespace wasn't changed while editing
+		if err = visitor.Visit(resource.RequireNamespace(cmdNamespace)); err != nil {
+			return preservedFile(err, file, cmd.Out())
+		}
+
+		// attempt to calculate a delta for merging conflicts
+		delta, err := jsonmerge.NewDelta(original, edited)
+		if err != nil {
+			glog.V(4).Infof("Unable to calculate diff, no merge is possible: %v", err)
+			delta = nil
+		} else {
+			delta.AddPreconditions(jsonmerge.RequireKeyUnchanged("apiVersion"))
+			results.delta = delta
+			results.version = defaultVersion
+		}
+
+		err = visitor.Visit(func(info *resource.Info) error {
+			data, err := info.Mapping.Codec.Encode(info.Object)
+			if err != nil {
+				return err
+			}
+			updated, err := resource.NewHelper(info.Client, info.Mapping).Update(info.Namespace, info.Name, false, data)
```

Comment:
I just confirmed that in the current version, I can call `kubectl edit pod pod1`, and copy paste the output of `kubectl get pod pod2` into the editor, and update pod2.

For a single object, it's easy to do a check before calling Update() here. For a List, it will be harder, we need to check the kind/name fields of all enclosed objects.